packages/e2e-tests/specs/editor/various/mentions.test.js

```diff
@@ -20,4 +31,27 @@ describe( 'autocomplete mentions', () => {
 		await page.keyboard.type( '.' );
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	it( 'should insert mention between two other words', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'Stuck in the middle with you.' );
+		await pressKeyTimes( 'ArrowLeft', 'you.'.length );
+		await page.keyboard.type( '@j' );
+		await page.waitForSelector( '.components-autocomplete__result' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( ' ' );
+		expect( await getEditedPostContent() ).toMatchSnapshot();
```

Comment:
We could experiment here and in the next test with [`toMatchInlineSnapshot`](https://jestjs.io/docs/expect#tomatchinlinesnapshotpropertymatchers-inlinesnapshot) to make tests more readable.

Reply:
@ntsekouras, have you tried using inline snapshots?

Reply:
Oh... I'm sorry Grzegorz but for some reason I thought it was suggestion to have in mind for `next tests` :). I read about it and just though that it was cool 😄 .

I created a small PR to update per your suggestion here: https://github.com/WordPress/gutenberg/pull/32727